pdfannots/printer/markdown.py

```diff
@@ -286,29 +288,52 @@ def emit_body(
 
         self._fmt_header_called = False
 
-        def fmt_header(name: str) -> str:
+        def fmt_header(name: str, level: int = 2) -> str:
+            """
+            A function that formats a header with a given name and level.
+
+            Parameters:
+                name (str): The name of the header.
+                level (int, optional): The level of the header. Defaults to 2.
+
+            Returns:
+                str: The formatted header.
+            """
             # emit blank separator line if needed
             prefix = '\n' if self._fmt_header_called else ''
             self._fmt_header_called = True
-            return prefix + "## " + name + "\n\n"
+            header = '#' * level
+            return prefix + header + " " + name + "\n"
 
         # Partition annotations into nits, comments, and highlights.
         nits = []
         comments = []
-        highlights = []
+        highlights = {} if self.group_highlights_by_color else []
         for a in document.iter_annots():
             if a.subtype in self.ANNOT_NITS:
                 nits.append(a)
             elif a.contents:
                 comments.append(a)
             elif a.subtype == AnnotationType.Highlight:
-                highlights.append(a)
+                if self.group_highlights_by_color:
+                    color = a.color.ashex()
+                    if color not in highlights:
+                        highlights[color] = []
+                    highlights[color].append(a)
+                else:
+                    highlights.append(a)
 
         for secname in self.sections:
             if highlights and secname == 'highlights':
                 yield fmt_header("Highlights")
-                for a in highlights:
-                    yield self.format_annot(a, document)
+                if self.group_highlights_by_color:
+                    for color, annots in highlights.items():
+                        yield fmt_header(f"Color: {color}", level=3)
```

Comment:
At this point it would be really nice to map RGB colours to names here... out of scope for this PR but I assume you've thought about it?

Reply:
It Would be very nice! I haven't looked into it in detail but I'd assume we aren't the first people facing this problem. I can have a look sometime this week. But I agree that this is out of scope for this PR.